code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1092,6 +1092,10 @@ var/global/list/damage_icon_parts = list()
 		O.name = "[index]"
 		O.icon = t_inhand_state
 		O.icon_state = t_state
+		if(I.color)
+			O.color = I.color
+		else
+			O.color = null // this prevents a bug where all inhands would be colored after holding a colored item
```

Comment:
So wait I'm confused
Wouldn't I.color = null OR a color value?
So isn't setting `O.color = I.color` always going to set it to null or a color?
So isn't the if else redundant?
Maybe I'm wrong